pkg/datapath/iptables/reconciler.go

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package iptables
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/datapath/tables"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/node"
+	"github.com/cilium/cilium/pkg/statedb"
+	"github.com/cilium/cilium/pkg/stream"
+)
+
+type desiredState struct {
+	firstInit    bool
+	installRules bool
+	devices      []*tables.Device
+	localNode    node.LocalNode
+}
+
+func reconciliationLoop(
+	ctx context.Context,
+	health cell.HealthReporter,
+	installRules bool,
+	localNodeStore *node.LocalNodeStore,
+	db *statedb.DB,
+	devices statedb.Table[*tables.Device],
+	updateRules func(ctx context.Context, state *desiredState) error,
+) error {
+	state := desiredState{
+		firstInit:    true,
+		installRules: installRules,
+	}
+
+	localNode, err := localNodeStore.Get(ctx)
```

Comment:
This isn't needed, `localNodeStore.Observable` emits the current state on subscribe, so you can just do `localNode := <-localNodeEvents` before the loop.